lib/src/kvs/tx.rs

```diff
@@ -1038,24 +1046,30 @@ impl Transaction {
 		}
 	}
 
-	pub(crate) fn clock(&self) -> Timestamp {
+	/// Clock retrieves the current timestamp which is fallible
+	/// It is used for unreliable ordering of events as well as
+	/// handling of timeouts. Operations that are not guaranteed to be correct.
+	/// But also allows for lexicographical ordering.
+	///
+	/// Public for tests, but we might not want to expose this
+	pub async fn clock(&self) -> Timestamp {
```

Comment:
We want this `pub(crate)` so as not to leak internal implementation not relevant to users
https://github.com/surrealdb/surrealdb/pull/2700#discussion_r1339033700